### PR TITLE
feat(explore): adhoc column formatting for Table chart

### LINKED ISSUE #10729: Improve Custom D3 Formatting for Source Columns on table viz type
**Is your feature request related to a problem? Please describe.**
Currently not all viz types allows users to user d3 formatting for their data.  For example viz type big number with trendline allows this, but table viz type doesn't. There is a hacky workaround, where users can specify this in the edit Datasource --> Calculated Column section, but it would be much better user experience, is they can do it in the explore itself.

Current **hacky** solution:
table viz 01

**Describe the solution you'd like**
Enable users to easily define their column type on table viz, specifically custom d3 formatting. 

**Describe alternatives you've considered**
Current solution works but is really hacky and non intuitive for users.

**Option 1:**  Let users edit when clicking on column name in Explore table viz. 
option 1

**Option 2:** Add a dropdown with formatting options in Edit Datasource --> Calcualted Column section
option 2
.. Similar is available on big trendline viz type and other viz types

trendline viz

**Option 3:** Enable both option (1) and (2), where option (1) saves the settings, but can also be visible and edited via option (2). This would provide users most flexibility.

**Additional context**
It would be great to also go through all the viz types and if there are other which don't allow this functionality, enable this for all viz types with as much consistency in the settings as possible across.

### PR DESCRIPTION
**Pull request by ktmud**
### SUMMARY

Add ad-hoc per-column formatting to table chart so users can change things like column width and d3 format after querying. 

Most logics are built in `superset-ui`, with this PR: https://github.com/apache-superset/superset-ui/pull/1019

The only change in `superset-frontend` is to pass chart state to controls.

Closes #10729 .

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

See https://github.com/apache-superset/superset-ui/pull/1019 for more screenshots.

### TEST PLAN

Manual testing

### ADDITIONAL INFORMATION

- [x] Has associated issue:
- [x] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/explore/components/ControlPanelsContainer.tsx [hunk 1]
```diff
@@ -128,17 +128,23 @@ export class ControlPanelsContainer extends React.Component<ControlPanelsContain
...
     // If the control item is not an object, we have to look up the control data from
     // the centralized controls file.
     // When it is an object we read control data straight from `config` instead
-    const controlData = {
+    let controlData = {
       ...config,
       ...controls[name],
       name,
     };
+    if (config.mapStateToProps?.length === 3) {
```
Comment (etr2460) [inline superset-frontend/src/explore/components/ControlPanelsContainer.tsx:142]: what does this mean? maybe a comment to show why it's necessary?

File: superset-frontend/src/explore/components/ControlPanelsContainer.tsx [hunk 2]
```diff
@@ -128,17 +128,23 @@ export class ControlPanelsContainer extends React.Component<ControlPanelsContain
...
 
     // If the control item is not an object, we have to look up the control data from
     // the centralized controls file.
     // When it is an object we read control data straight from `config` instead
-    const controlData = {
+    let controlData = {
       ...config,
       ...controls[name],
       name,
     };
```
Comment (etr2460) [inline superset-frontend/src/explore/components/ControlPanelsContainer.tsx:141]: actually, why not just

```suggestion
    const controlData = {
      ...config,
      ...controls[name],
      ...(config.mapStateToProps?.length === 3 ? config.mapStateToProps(exploreState, controls[name], chart) : {}),
      name,
    };
```

File: superset-frontend/src/explore/components/ControlPanelsContainer.tsx (hunk 1 above)
Comment (ktmud) [reply superset-frontend/src/explore/components/ControlPanelsContainer.tsx:142]: I updated the comments

File: superset-frontend/src/explore/components/ControlPanelsContainer.tsx (hunk 2 above)
Comment (ktmud) [reply superset-frontend/src/explore/components/ControlPanelsContainer.tsx]: Done

**Review by etr2460 — APPROVED** [review]
lgtm, and thanks for bumping the package versions to pull in my fix too!

**Comment by ktmud** [comment]
It seems emotion's `css={{ ... }}` prop is not picked up when used inside a `node_modules` library. Will investigate and find a fix.

**Pull request merged**